src/sentry/workflow_engine/handlers/detector/stateful.py

```diff
@@ -279,30 +410,53 @@ def extract_group_values(
         """
         pass
 
-    def build_fingerprint(self, group_key: DetectorGroupKey) -> list[str]:
+    def build_group_fingerprint(self, group_key: DetectorGroupKey) -> list[str]:
         """
         Builds a fingerprint to uniquely identify a detected issue
 
         TODO - Take into account the data source / query that triggered the detector,
         we'll want to create a new issue if the query changes.
         """
-        return [f"{self.detector.id}{':' + group_key if group_key is not None else ''}"]
+        return [self.build_key_for_group(group_key)]
+
+    def build_key_for_group(self, group_key: DetectorGroupKey, postfix: str | None = None) -> str:
+        """
+        Builds a key for the given group key. This is used to store the state of the detector in Redis.
+        """
+        group_postfix = f"{group_key if group_key is not None else ''}"
```

Comment:
This is all the same code as in `_build_key` on line 125, can it be consolidated?

Reply:
Err idk if I selected the lines correctly but lines 426 - 429 are the same